### DISCUSSION TITLE
Aptos testnet faucet api spec

### DISCUSSION THREAD

**Discussion by gftea** [aptos-labs/aptos-developer-discussions#245] [Questions]
### Describe your question in detail.

The facuet api on testnet is not correct, it seems, spec is out of date. Can you help to clarify faucet api spec?
Also when I looking into `FaucetClient` in CLI it use another url spec. Which one is correct?
```
    /// Fund an account with the given amount.
    pub async fn fund(&self, address: AccountAddress, amount: u64) -> Result<()> {
        let mut url = self.faucet_url.clone();
        url.set_path("mint");
        let query = format!("auth_key={}&amount={}&return_txns=true", address, amount);
        url.set_query(Some(&query));
```

### What have you tried or looked at?  Or how can we reproduce the error?

e.g.
1. heathcheck api, I call it accordign to https://faucet.testnet.aptoslabs.com/spec#/operations/root, and it returns 404, if I remove `/v1`, then it returns `tap:ok`, indicating the spec is incorrect vs deployed one.
![image](https://github.com/aptos-labs/aptos-developer-discussions/assets/1705787/47014ad7-01b8-4341-ae98-c3fd2dad1caf)

2. fund api https://faucet.testnet.aptoslabs.com/spec#/operations/fund
```python
import requests

url = "https://faucet.testnet.aptoslabs.com/v1/fund"

payload = {
    "amount": 0,
    "auth_key": "0xe5c2447ee3376ad4b20e9b957aaf1a1946cbabf3bcee12290832754effeada54",
    "address": "0xe5c2447ee3376ad4b20e9b957aaf1a1946cbabf3bcee12290832754effeada54",
    "pub_key": "0x0dffda3788d6c55c13b22827e056785d20538801cd254eff840603e3861523c2"
}
headers = {
    "Content-Type": "application/json",
    "Accept": "application/json"
}

response = requests.post(url, json=payload, headers=headers)
print(response.json())
```
It returns error
`{'message': 'not found', 'error_code': 'WebFrameworkError', 'rejection_reasons': [], 'txn_hashes': []}`

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

REST API

**Comment by gregnazario** [comment]
I just noticed this yesterday with some tests I was running as well by mistake.  Thanks for the callout!

Should be fixed to remove the `/v1`, this PR I believe will do it https://github.com/aptos-labs/aptos-core/pull/13444

**Reply by gftea** [reply]
Hi it seems not only /v1, if I use `"https://faucet.testnet.aptoslabs.com/fund", it also return error

**Reply by gregnazario** [reply]
Hi, need to use a POST request, it should work then
